h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1313,14 +1335,25 @@ else if (_parms._weights_column != null && _weights != null && !_weights.isBinar
     // Toss out extra columns, complain about missing ones, remap categoricals
     Frame va = _parms.valid();  // User-given validation set
     if (va != null) {
-      _valid = adaptFrameToTrain(va, "Validation Frame", "_validation_frame", expensive);
+      //  TODO    we are using in `adaptFrameToTrain`   _train._names which was not yet encoded
+      Frame adaptedValid = adaptFrameToTrain(va, "Validation Frame", "_validation_frame", expensive);
+      setValid(adaptedValid);
```

Comment:
is it to be extended in future commit? don't see any change here otherwise.

Reply:
@sebhrusen artefact from "enum" approach". Will fix